File: examples/nodejs/05_time_series.js
const Redis = require("ioredis");

const redis = new Redis({ host: "localhost", port: 6379 });

async function main() {
    console.log("--- 05 Time Series ---");
    const key = "server:cpu:load";

    const now = Date.now();

    // TS.ADD
    console.log("Adding CPU load metrics...");
    await redis.call("TS.ADD", key, now, 45.0);
    await redis.call("TS.ADD", key, now + 1000, 50.5);
    await redis.call("TS.ADD", key, now + 2000, 48.2);

    // TS.RANGE
    console.log("Querying metrics...");
    const data = await redis.call("TS.RANGE", key, now, now + 5000);
    console.log("Data:", data);

    redis.disconnect();
}

main();

File: examples/python/05_time_series.py
import redis
import time

# Connect to Zedis
r = redis.Redis(host='localhost', port=6379, decode_responses=True)

print("--- 05 Time Series ---")

ts_key = "sensor:temp:room1"

# TS.ADD (Key, Timestamp, Value)
start_time = int(time.time() * 1000)

print(f"Adding data points to {ts_key}...")
r.execute_command('TS.ADD', ts_key, start_time, 22.5)
r.execute_command('TS.ADD', ts_key, start_time + 1000, 23.0)
r.execute_command('TS.ADD', ts_key, start_time + 2000, 23.2)
r.execute_command('TS.ADD', ts_key, start_time + 3000, 22.8)

# TS.RANGE (Key, Start, End)
print("\nQuerying range:")
# Query from start_time to start_time + 5000
data = r.execute_command('TS.RANGE', ts_key, start_time, start_time + 5000)
print(data)
